sklearn/utils/__init__.py

```diff
@@ -225,6 +226,21 @@ def safe_indexing(X, indices, axis=0):
         )
 
 
+def _array_indexing(array, key, axis=0):
+    """Index an array consistently across NumPy version."""
+    if axis not in (0, 1):
+        raise ValueError(
+            "'axis' should be either 0 (to index rows) or 1 (to index "
+            " column). Got {} instead.".format(axis)
+        )
+    if np_version < (1, 13) or issparse(array):
```

Comment:
boolean array-like seems to work as expected for numpy 1.12

```suggestion
    if np_version < (1, 12) or issparse(array):
```

Reply:
> boolean array-like seems to work as expected for numpy 1.12

Do you mean 1.13?

Reply:
I meant 1.12:

```py
import numpy as np
X = np.array([1, 3])
```

### 1.13.1

```py
X[[True, False]]
# array([1])
```

### 1.12.1

```py
X[[True, False]]
# array([1])
```

### 1.11.3

```py
# 1.11.3
X[[True, False]]
# FutureWarning: in the future, boolean array-likes will be 
# handled as a boolean array index
# array([3, 1])
```

Reply:
Oh I see

Reply:
Oh I just read in the what's new of NumPy without actually trying it. `1.12` it is then.